code/modules/cooking/machinery/foodcart.dm

```diff
@@ -0,0 +1,147 @@
+/obj/machinery/food_cart
+	name = "food cart"
+	desc = "A compact unpackable mobile cooking stand. Wow! When unpacked, it reminds you of those greasy gamer setups some people on NTNet have."
+	icon = 'icons/obj/machinery/cooking_machines.dmi'
+	icon_state = "foodcart"
+	density = TRUE
+	anchored = FALSE
+	stat = POWEROFF
+	use_power = POWER_USE_OFF
+	req_access = list(ACCESS_KITCHEN)
+	var/unpacked = FALSE
+	// Stuff that's gonna make up the food stand when unpacked. Self-explanatory.
+	var/obj/machinery/appliance/cooker/grill/stand/cart_griddle
+	var/obj/machinery/smartfridge/foodheater/stand/cart_smartfridge
+	var/obj/structure/table/reinforced/cart_table
+	var/obj/effect/food_cart_stand/cart_tent
+	var/list/packed_things
+
+/obj/machinery/food_cart/Initialize(mapload)
+	. = ..()
+	cart_griddle = new(src)
+	cart_smartfridge = new(src)
+	cart_table = new(src)
+	cart_tent = new(src)
+	packed_things = list(cart_table, cart_smartfridge, cart_tent, cart_griddle) //middle, left, left, right
+	RegisterSignal(cart_griddle, COMSIG_QDELETING, PROC_REF(lost_part))
+	RegisterSignal(cart_smartfridge, COMSIG_QDELETING, PROC_REF(lost_part))
+	RegisterSignal(cart_table, COMSIG_QDELETING, PROC_REF(lost_part))
+	RegisterSignal(cart_tent, COMSIG_QDELETING, PROC_REF(lost_part))
+
+/obj/machinery/food_cart/Destroy()
+	QDEL_NULL(cart_griddle)
+	QDEL_NULL(cart_smartfridge)
+	QDEL_NULL(cart_table)
+	QDEL_NULL(cart_tent)
+	packed_things.Cut()
+	return ..()
+
+/obj/machinery/food_cart/get_examine_text(mob/user, distance, is_adjacent, infix, suffix)
+	. = ..()
+	if(stat & BROKEN)
+		return
+	if(cart_griddle.stat & BROKEN)
+		. += SPAN_WARNING("The stand's [SPAN_BOLD("grill")] is completely broken!")
+	else
+		. += SPAN_NOTICE("The stand's [SPAN_BOLD("grill")] is intact.")
+	. += SPAN_NOTICE("The stand's [SPAN_BOLD("fridge")] seems fine.") //weirdly enough, these fridges don't break
+	. += SPAN_NOTICE("The stand's [SPAN_BOLD("table")] seems fine.")
+
+/obj/machinery/food_cart/proc/pack_up() // Retract the structures into the food cart
+	if(!unpacked)
+		return
+	visible_message(SPAN_NOTICE("[src] retracts all of it's unpacked components."))
+	for(var/obj/object as anything in packed_things)
+		UnregisterSignal(object, COMSIG_MOVABLE_MOVED)
+		object.forceMove(src)
+	if(!(cart_griddle?.stat & BROKEN)) // Don't draw power if it's packed inside
+		cart_griddle?.stat = POWEROFF
+		cart_griddle?.use_power = POWER_USE_OFF
+		cart_griddle?.update_icon()
+	if(!(cart_smartfridge?.stat & BROKEN))
+		cart_smartfridge.use_power = POWER_USE_OFF
+	anchored = FALSE
+	unpacked = FALSE
+
+/obj/machinery/food_cart/proc/unpack(mob/user) // Deploy the structures inside the food cart
```

Comment:
https://github.com/Aurorastation/Aurora.3/blob/master/.github/guides/STYLE.md#documenting-a-proc

That isn't a DMDoc